Offline build option is not passed to inner BWC builds

At least I think this is what's happening. I am trying to run a full cluster restart IT offline.

I have a checkout of `master` (6f32f7) and of a recent 6.4 commit (7d09fb) for the BWC snapshot, but trying to run the test fails during the BWC build because it's unable to download the ML snapshot.

The lack of WiFi makes it hard to report this properly, but I can follow up when I have connectivity if needed.

This seems to reproduce the issue I saw, if run while offline:

```
$ git rev-parse HEAD
6f32f7138db47f503974bd7ce71f32449289a3c7
$ git rev-parse 6.x
22cb973bfd3ca6dbc52e21ebc1b7effba6274624
$ rm -rf distribution/bwc/next-minor-snapshot/build/ && git clean -dxf
$ ./gradlew --offline :qa:full-cluster-restart:v6.5.0-SNAPSHOT#oldClusterTest -Dtests.bwc.refspec.6.x=22cb973bfd3ca6dbc52e21ebc1b7effba6274624 2>&1 >testoutput.log
[...snip...]
  [bwc] FAILURE: Build failed with an exception.
  [bwc]
  [bwc] * Where:
  [bwc] Build file '/Users/davidturner/src/elasticsearch-master/distribution/bwc/next-minor-snapshot/build/bwc/checkout-6.x/x-pack/plugin/ml/cpp-snapshot/build.gradle' line: 18
  [bwc]
  [bwc] * What went wrong:
  [bwc] Execution failed for task ':x-pack:plugin:ml:cpp-snapshot:downloadMachineLearningSnapshot'.
  [bwc] > java.net.UnknownHostException: prelert-artifacts.s3.amazonaws.com
[...snip...]
```

--- a/distribution/bwc/build.gradle
+++ b/distribution/bwc/build.gradle
@@ -179,6 +179,9 @@ subprojects {
     } else {
       executable new File(checkoutDir, 'gradlew').toString()
     }
+    if (gradle.startParameter.isOffline()) {
+      args "--offline"
+    }
     for (String dir : projectDirs) {
       args ":${dir.replace('/', ':')}:assemble"
     }

